core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -161,6 +162,25 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
     this
   }
 
+  private final val avroNamespace = "avro.schema."
+
+  /**
+   * Use Kryo serialization and register the given set of Avro schemas so that the generic
+   * record serializer can decrease network IO
+   */
+  def registerAvroSchemas(schemas: Schema*): SparkConf = {
+    schemas.foldLeft(this) { (conf, schema) =>
```

Comment:
Why do you need a fold here?  This seems confusing.  Why not just loop over the schemas and call `conf.set()` for each?